### PR TITLE
[doc] Improve Pulsar Security Extending

### PR DESCRIPTION
**Pull request by Monica-zy**
Motivation
Improve the language and the overall descriptive style of the Pulsar Security document (security-extending section): http://pulsar.apache.org/docs/en/next/security-extending/

Modifications
Adjust the tone, personal pronouns, voice also some typo errors of some sentences in the document.

### CODE REVIEW

**Comment by Monica-zy** [comment]
@Jennifer88huang Please review~

File: site2/docs/security-extending.md [hunk 1]
```diff
@@ -27,35 +23,30 @@ PulsarClient client = PulsarClient.builder()
...
-For reference, there are 2 interfaces to implement on the client side:
+For reference, you can use 2 interfaces to implement on the client side:
  * `Authentication` -> http://pulsar.apache.org/api/client/org/apache/pulsar/client/api/Authentication.html
  * `AuthenticationDataProvider` -> http://pulsar.apache.org/api/client/org/apache/pulsar/client/api/AuthenticationDataProvider.html
 
 
-This in turn will need to provide the client credentials in the form of `org.apache.pulsar.client.api.AuthenticationDataProvider`. This will leave
-the chance to return different kinds of authentication token for different
-type of connection or by passing a certificate chain to use for TLS.
+This in turn needs to provide the client credentials in the form of `org.apache.pulsar.client.api.AuthenticationDataProvider`. This leaves the chance to return different kinds of authentication token for different type of connection or by passing a certificate chain to use for TLS.
```
Comment (Jennifer88huang) [inline site2/docs/security-extending.md]: maybe "a different type" or "different types"?

Comment (Jennifer88huang) [reply site2/docs/security-extending.md]: refine the sentence "This leaves the chance to ..."

File: site2/docs/security-extending.md [hunk 2]
```diff
@@ -92,22 +83,18 @@ public interface AuthenticationProvider extends Closeable {
...
+Authorization is the operation that checks whether a particular "role" or "principal" has a permission to perform a certain operation.
 
-By default, Pulsar provides an embedded authorization, though it's possible to
-configure a different one through a plugin.
+By default, Pulsar provides an embedded authorization, though configuring a different one through a plugin is also an alternative choice.
 
-To provide a custom provider, one needs to implement the
- `org.apache.pulsar.broker.authorization.AuthorizationProvider` interface, have this class in the
- Pulsar broker classpath and configure it in `conf/broker.conf`:
+To provide a custom provider, you need to implement the `org.apache.pulsar.broker.authorization.AuthorizationProvider` interface, have this class in the Pulsar broker classpath and configure it in `conf/broker.conf`:
```
Comment (Jennifer88huang) [inline site2/docs/security-extending.md]: refine "have this class..."

**Comment by Monica-zy** [comment]
run java8 tests
run integration tests

**Comment by Monica-zy** [comment]
run integration tests

**Comment by Monica-zy** [comment]
run integration tests

**Comment by Monica-zy** [comment]
run integration tests

**Comment by Monica-zy** [comment]
run integration tests

**Comment by Monica-zy** [comment]
run integration tests

**Pull request merged**